src/main/java/se/liu/ida/hefquin/engine/queryplan/utils/PhysicalPlanFactory.java

```diff
@@ -721,11 +734,17 @@ public static boolean checkUnaryOpApplicableToUnionPlan( final PhysicalPlan unio
 		for ( int i = 0; i < unionPlan.numberOfSubPlans(); i++ ) {
 			final PhysicalPlan subPlan = unionPlan.getSubPlan(i);
 			final PhysicalOperator subRootOp = subPlan.getRootOperator();
-			if ( !(subRootOp instanceof PhysicalOpRequest || subRootOp instanceof PhysicalOpFilter) ) {
+			if ( !(subRootOp instanceof PhysicalOpRequest || subRootOp instanceof PhysicalOpFilter || subRootOp instanceof PhysicalOpLocalToGlobal ) ) {
 				return false;
 			}
 
-			if ( subRootOp instanceof PhysicalOpFilter){
+			if ( subRootOp instanceof PhysicalOpLocalToGlobal ){
+				if ( !( subPlan.getSubPlan(0) instanceof PhysicalOpRequest || subPlan.getSubPlan(0) instanceof PhysicalOpFilter) ){
```

Comment:
Shouldn't there be an additional condition to check, if the operator under the L2G operator is a filter operator,   that then under that filter there must be a request.